### ISSUE TITLE
Duplicate spaces (/msg)

### ISSUE THREAD

**Issue by ghost** [rocketman10404/tabbychat#18]
TabbyChat is sending double spaces when replying to a personal message.

TabbyChat GUI: /msg sourcemaker test
Server console: /msg sourcemaker  test

Replaced whitespaces with #:
GUI: /msg#sourcemaker#test
Console: /msg#sourcemaker##test

Essentials is not able to handle two whitespaces after the name and will abort its functionality.

**Comment by ghost** [comment]
I made a server-side (bukkit) plugin to remove all duplicate whitespaces in all commands.
Maybe its a hotfix until this issue is processed ;-)

https://dl.dropboxusercontent.com/u/20809836/ChunkTools.jar
(source code is included)

**Comment by rocketman10404** [comment]
I'm unfortunately unable to reproduce this issue.  I've tested vanilla and bukkit with essentials, both by manually typing in a "/msg" command, and also using a defined tab command prefix, and in all cases only space character exists in both the GUI (as you mentioned) and the server console.

Could I possibly be missing some other prequisites to producing this behavior?

**Comment by rocketman10404** [comment]
## Reported via e-mail, reposting here for reference

Love the mod, great work
I found a bug in the 1.4.7 version, it probably exists in the other version since the underlying issue is in the code on github.
The symptoms:
When making a 1 line reply to any private message it adds an extra space between the command (/msg <name>) and the message and the server complains and doesnt send the message.
This DOESNT happen when you send a multiline message.
This also happens on any other channel when you add commands prefixes but I noticed it first on a PM channel.

I had a look at your chat string formatting suspecting that there is a for loop that adds space differntly then the case of a single line which would cause this and I found just that:

tabbychat / acs / tabbychat / util / 
BackgroundChatThread.java 
line 59:     mc.thePlayer.sendChatMessage(cmdPrefix + sendPart.toString());

Here I think you should trim the sendPart just like you have done at line 47 in order to avoid double space.

This explains why it works properly when its processing a multiline but not when its processing a single line.
I probably would have added the space here instead of in the cmdPrefix myself and just trim both but either way works.

I hope this was helpful for you

-IGN Pandypants

**Comment by rocketman10404** [comment]
Ok, I think I see it now.  My mistake for not looking into this closely enough before.  Either way, with a vanilla server or bukkit (with or w/o essentials) I'm still not getting kicked or anything, and everything displays as one space.  However, when I step through this with a debugger, I can definitely reproduce two spaces between the command and the message.

The problem should only occur when there's a defined chat command prefix used in the message - such as sending a PM from an auto-created PM tab.  The specific problem takes place on lines 26-27 of BackgroundChatThread.java:

```
cmdPrefix = this.knownPrefix.trim() + " ";
this.sendChat = this.sendChat.substring(this.knownPrefix.length());
```

Here we take the knownPrefix (which is always trimmed of whitespace) which should store the tab's command prefix, and we add the whitespace to it, storing that result in a local `cmdPrefix` variable.  Then, we remove the length of knownPrefix, without the whitespace, from the rest of the message.  Therefore, we've added a whitespace to the separate prefix, and retained the same whitespace in the message.

I'll just add a trim at the end of line 27.  The lack of a trim at line 59 is unrelated to this issue, although having one may have hidden the problem.  Regardless, adding one here as well should help remove any trailing whitespaces left as a result of line 56.  Before that, I need to figure if this is related to either #21 or #22 first, so I can verify that they'd be resolved.

Regardless, thanks for both of your help, should be fixed in the next release.

**Comment by rocketman10404** [comment]
Fixed in 6d719bd99af8fbeb4ece19359950c7b94652409d
